src/mscorlib/corefx/System/IO/PathHelper.Windows.cs

```diff
@@ -271,119 +268,124 @@ private static string TryExpandShortFileName(StringBuffer outputBuffer, string o
             //
             // Note that we will never get \??\ here as GetFullPathName() does not recognize \??\ and will return it as C:\??\ (or whatever the current drive is).
 
-            uint rootLength = PathInternal.GetRootLength(outputBuffer);
-            bool isDevice = PathInternal.IsDevice(outputBuffer);
+            int rootLength = PathInternal.GetRootLength(ref outputBuffer);
+            bool isDevice = PathInternal.IsDevice(ref outputBuffer);
 
-            StringBuffer inputBuffer = null;
-            bool isDosUnc = false;
-            uint rootDifference = 0;
-            bool wasDotDevice = false;
-
-            // Add the extended prefix before expanding to allow growth over MAX_PATH
-            if (isDevice)
+            StringBuffer inputBuffer = new StringBuffer(0);
```

Comment:
I am not sure whether we want to have `StringBuffer.Empty`. Having default constructors for structs would be best in this case. I will leave it as is for now. We will hit the pool, but it will return quickly because of it has optimization for empty arrays as well.